typeid/typeid-go/typeid.go

```diff
@@ -62,96 +67,143 @@ func (tid TypeID) UUIDBytes() []byte {
 }
 
 // UUID decodes the TypeID's suffix as a UUID and returns it as a hex string
-func (tid TypeID) UUID() string {
+func (tid TypeID[P]) UUID() string {
 	return uuid.FromBytesOrNil(tid.UUIDBytes()).String()
 }
 
 // From returns a new TypeID with the given prefix and suffix.
 // If suffix is the empty string, a random suffix will be generated.
 // If you want to create an id without a prefix, pass an empty string as the prefix.
-func From(prefix string, suffix string) (TypeID, error) {
-	if err := validatePrefix(prefix); err != nil {
-		return Nil, err
+func From(prefix string, suffix string) (TypeID[AnyPrefix], error) {
+	return from(AnyPrefix(prefix), suffix)
+}
+
+// FromPrefix returns a new TypeID with the given prefix and a random suffix.
+// If you want to create an id without a prefix, pass an empty string.
+func FromPrefix(prefix string) (TypeID[AnyPrefix], error) {
```

Comment:
I might call this `NewWithPrefix` or `WithPrefix`

`from` implies you are creating the ID from the prefix which is not 100% true. (See how `From` and `FromString` are different)